packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -382,10 +383,18 @@ export class ExternalRequest {
       }
       const display = linkedTable.primaryDisplay
       for (let key of Object.keys(row[relationship.column])) {
-        const related: Row = row[relationship.column][key]
+        let relatedRow: Row = row[relationship.column][key]
+        // add this row as context for the relationship
+        for (let col of Object.values(linkedTable.schema)) {
+          if (col.type === FieldType.LINK && col.tableId === table._id) {
+            relatedRow[col.name] = [row]
```

Comment:
Is this right? Is the foreign row an array even if it's a simple link?

Reply:
Yeap - it always uses an array, even for a simple link, thats just our relationship system, for a one relation, you still need to do `relationship[0]`.

This could change - but its all based on the way we handle relationships in our internal tables.